src/challengePaths/2021/YouRobot.ts

```diff
@@ -0,0 +1,139 @@
+import { round, runChoice, Slot, Stat, visitUrl } from "kolmafia";
+import { have } from "../../lib.js";
+import { get } from "../../property.js";
+import { $path, $skill, $slot, $stat } from "../../template-string.js";
+
+export const path = $path`You, Robot`;
+
+type Upgrade = "robot_shirt" | "robot_energy" | "robot_potions";
+
+/**
+ * Runs one turn scavenging in the scrapheap
+ */
+export function scavenge(): void {
+  visitUrl("place.php?whichplace=scrapheap&action=sh_scrounge");
+}
+
+/**
+ * Spends one turn collecting energy in the scrapheap
+ */
+export function collectEnergy(): void {
+  visitUrl("place.php?whichplace=scrapheap&action=sh_getpower");
+}
```

Comment:
These two functions are very similar. They should have more similar docstrings.

I'd also have them return a `boolean` that essentially says "did we successfully do this"--maybe add checks for if the user is in YR and has an adventure. Alternately, you could return the total amount of energy or scavenge parts? There is a meaningful value you can return and you should return it.